src/Tools/BuildValidator/LocalReferenceResolver.cs

```diff
@@ -111,7 +111,7 @@ public bool TryResolveReferences(ImmutableArray<MetadataReferenceInfo> reference
                     filePath: filePath,
                     properties: new MetadataReferenceProperties(
                         kind: MetadataImageKind.Assembly,
-                        aliases: reference.ExternAliases,
+                        aliases: reference.ExternAlias is null ? ImmutableArray<string>.Empty : ImmutableArray.Create(reference.ExternAlias),
```

Comment:
Isn't the property `ExternAlias` of type `string`, meaning this expression will create an array of a single alias even if multiple comma-separated aliases are present?

Reply:
This code will maintain the pattern of a 1:1 relationship between extern alias and the reference. That is how the compiler actually represents extern aliases internally. The fact that the property back on `MetadataReference` has a collection of aliases is a red herring. It may be useful in API driven scenarios (have not checked) but in the command line scenario it is always a 1:1 relationship. 

It is a very confusing detail and the majority of my time in this change was understanding this relationship.